mempool/src/shared_mempool/types.rs

```diff
@@ -41,6 +41,28 @@ where
     pub subscribers: Vec<UnboundedSender<SharedMempoolNotification>>,
 }
 
+impl<V: TransactionValidation + 'static> SharedMempool<V> {
+    pub fn new(
+        mempool: Arc<Mutex<CoreMempool>>,
+        config: MempoolConfig,
+        network_senders: HashMap<NetworkId, MempoolNetworkSender>,
+        db: Arc<dyn DbReader<DpnProto>>,
+        validator: Arc<RwLock<V>>,
+        peer_manager: Arc<PeerManager>,
+        subscribers: Vec<UnboundedSender<SharedMempoolNotification>>,
+    ) -> Self {
+        SharedMempool {
+            mempool,
+            config,
+            network_senders,
+            db,
+            validator,
+            peer_manager,
+            subscribers,
+        }
+    }
+}
+
```

Comment:
why though?  I personally like in place, since it helps link up labels and types more readily. I feel constructors make sense if you actually need to do extra things.

Reply:
I prefer this way cause it gets rid of all the names, especially since the types still have to match.  But, I can just drop this commit.

Reply:
suuuuure ;)

btw please don't resolve unless it has been some amount of time or the commenter resolves as these comments can get easily lost.